README_local_MacOS.md

```diff
@@ -0,0 +1,122 @@
+# Example Local Dev Environment Setup
+
+## MacOS
+
+* `Go` version from
+  [go.mod](https://github.com/gravitational/teleport/blob/master/go.mod#L3)
+  
+  * Follow [official instructions](https://go.dev/doc/install) to install `Go`
+    * **On an M1 Mac, download ARM64 installer from https://go.dev/dl/**
+    * Download the installer for `<version from go.mod>`  
+    * After installing, don't forget to `export PATH="/usr/local/go/bin:$PATH"` in `~/.zprofile`
+    * If you need other go versions, see https://go.dev/doc/manage-install
+      * You will need to add `export PATH="$HOME/go/bin:$PATH"` to the `~/.zprofile`
+
+  * Or install required version of `Go` with homebrew:
+
+  ```shell
+  # if we are not on the latest, you might need to
+  # $ brew install go@<version from go.mod>, i.e. 1.16
+  #
+  # check which version will be installed by running
+  # $ brew info go
+  
+  $ brew install go
```

Comment:
I would drop the "$" from the snippets so it's easier to copy and paste the commands.

Reply:
This is highly subjective. The `$` make the user pay attention to what exactly is being copy-pasted and I personally like it that way.

Reply:
Cleaned up `$` from command snippets in README.md and the MacOS instructions.